runtime/compiler/x/codegen/J9AheadOfTimeCompile.cpp

```diff
@@ -261,12 +261,7 @@ uint8_t *J9::X86::AheadOfTimeCompile::initializeAOTRelocationHeader(TR::Iterated
             TR_OpaqueMethodBlock *method = resolvedMethod->getPersistentIdentifier();
             TR_OpaqueClassBlock *thisClass = guard->getThisClass();
             uint16_t methodID = symValManager->getIDFromSymbol(static_cast<void *>(method));
-            uint16_t receiverClassID = 0;
-            if (relocation->getTargetKind() == TR_InlinedInterfaceMethodWithNopGuard ||
-                relocation->getTargetKind() == TR_InlinedAbstractMethodWithNopGuard)
-               {
-               receiverClassID = symValManager->getIDFromSymbol(static_cast<void *>(thisClass));
-               }
+            uint16_t receiverClassID = symValManager->getIDFromSymbol(static_cast<void *>(thisClass));
```

Comment:
At first I wasn't sure how this would work, since there is no change to pass in `SymOptional` to the associated `getClassFromID` in the load infra code, but then I saw

https://github.com/eclipse/openj9/blob/2f2d9a6df07350f2045e6f240a5190f41c238eb6/runtime/compiler/runtime/RelocationRecord.cpp#L2230-L2233

so it all makes sense.